### PR TITLE
Add optional -emit-symbol-graph output when emitting modules

### PR DESCRIPTION
**Pull request by bitjammer**
rdar://71497047

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

File: lib/Frontend/Frontend.cpp [hunk 1]
```diff
@@ -165,6 +165,19 @@ SerializationOptions CompilerInvocation::computeSerializationOptions(
...
+  if (opts.EmitSymbolGraph) {
+    if (!opts.SymbolGraphOutputDir.empty()) {
+      serializationOpts.SymbolGraphOutputDir = opts.SymbolGraphOutputDir;
+    } else {
+      serializationOpts.SymbolGraphOutputDir = serializationOpts.OutputPath;
+    }
+    SmallString<256> OutputDir(serializationOpts.SymbolGraphOutputDir);
+    llvm::sys::fs::make_absolute(OutputDir);
+    llvm::sys::path::remove_filename(OutputDir);
+    serializationOpts.SymbolGraphOutputDir = OutputDir.str().str();
```
Comment (QuietMisdreavus) [inline lib/Frontend/Frontend.cpp:178]: Is there a specific reason `.str()` is called twice here?

Comment (bitjammer) [reply lib/Frontend/Frontend.cpp:178]: The first creates a `StringRef`, and the second copies into a `std::string`. It's just a naming coincidence.

Comment (QuietMisdreavus) [reply lib/Frontend/Frontend.cpp:178]: Interesting, thanks for the explanation!

**Review by QuietMisdreavus — APPROVED** [review]
Looks good, thanks so much!

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Pull request merged**